src/collectors/apps.plugin/README.md

```diff
@@ -140,48 +160,39 @@ group: process1 process2 ...
-undesirable, the line `other: *` should be appended to the `apps_groups.conf`.
+For each process given, all its sub-processes will be grouped, not just the process matched.
 
 The process names are the ones returned by:
 
--   `ps -e` or `cat /proc/PID/stat`
--   in case of substring mode (see below): `/proc/PID/cmdline`
+-   **comm**: `ps -e` or `cat /proc/{PID}/stat`
+-   **cmdline**: in case of substring mode (see below): `/proc/{PID}/cmdline`
+
+On Linux **comm** is limited to just a few characters. `apps.plugin` attempts to find the entire
+**comm** name by looking for it at the **cmdline**. When this succeeds, the entire process name
+is available, otherwise the shortened one is used.
 
 To add process names with spaces, enclose them in quotes (single or double)
 example: `'Plex Media Serv'` or `"my other process"`.
 
-You can add an asterisk `*` at the beginning and/or the end of a process:
+You can add asterisks (`*`) to provide a pattern:
 
--   `*name` _suffix_ mode: will search for processes ending with `name` (at `/proc/PID/stat`)
--   `name*` _prefix_ mode: will search for processes beginning with `name` (at `/proc/PID/stat`)
--   `*name*` _substring_ mode: will search for `name` in the whole command line (at `/proc/PID/cmdline`)
+-   `*name` _suffix_ mode: will match a **comm** ending with `name`.
+-   `name*` _prefix_ mode: will match a **comm** beginning with `name`.
+-   `*name*` _substring_ mode: will search for `name` in **cmdline**.
 
-If you enter even just one _name_ (substring), `apps.plugin` will process
-`/proc/PID/cmdline` for all processes (of course only once per process: when they are first seen).
+Asterisk may appear in the middle of `name` (like `na*me`), without affecting what is being
+matched (**comm** or **cmdline**). 
 
 To add processes with single quotes, enclose them in double quotes: `"process with this ' single quote"`
 
 To add processes with double quotes, enclose them in single quotes: `'process with this " double quote'`
 
-If a group or process name starts with a `-`, the dimension will be hidden from the chart (cpu chart only).
-
-If a process starts with a `+`, debugging will be enabled for it (debugging produces a lot of output - do not enable it in production systems).
-
-You can add any number of groups. Only the ones found running will affect the charts generated.
-However, producing charts with hundreds of dimensions may slow down your web browser.
-
 The order of the entries in this list is important: the first that matches a process is used, so put important
-ones at the top. Processes not matched by any row, will inherit it from their parents or children.
-
-The order also controls the order of the dimensions on the generated charts (although applications started
-after apps.plugin is started, will be appended to the existing list of dimensions the `netdata` daemon maintains).
+ones at the top. Processes not matched by any row, will inherit it from their parents.
```

Comment:
```suggestion
Processes not matched by any row, will inherit it from their parents.
```

@ktsaou here I remove the remnant of the previous sentence, but I got a question:

> Processes not matched by any row, will inherit **it** from their parents.

What does "it" represent here? Could it be referring to the snippet you removed:

```md
If a group or process name starts with a `-`, the dimension will be hidden from the chart (cpu chart only).

If a process starts with a `+`, debugging will be enabled for it (debugging produces a lot of output - do not enable it in production systems).

You can add any number of groups. Only the ones found running will affect the charts generated.
However, producing charts with hundreds of dimensions may slow down your web browser.
```

If it does, then probably the whole line can be deleted.

Reply:
Their group.
So we have PIDs (processes) which we group together in a meaningful way. So, PIDs not matches by any row, will inherit their group from their parents.